NimbleTests/AsynchronousTest.swift

```diff
@@ -63,22 +76,77 @@ class AsyncTest: XCTestCase {
                 done()
             }
         }
-        // "clear" runloop to ensure this test doesn't poison other tests
-        NSRunLoop.mainRunLoop().runUntilDate(NSDate().dateByAddingTimeInterval(0.2))
     }
 
     func testWaitUntilDetectsStalledMainThreadActivity() {
-        dispatch_async(dispatch_get_main_queue()) {
-            NSThread.sleepForTimeInterval(2.0)
+        let msg = "-waitUntil() timed out but was unable to run the timeout handler because the main thread is unresponsive (0.5 seconds is allow after the wait times out). Conditions that may cause this include processing blocking IO on the main thread, calls to sleep(), deadlocks, and synchronous IPC. Nimble forcefully stopped run loop which may cause future failures in test run."
+        failsWithErrorMessage(msg) {
+            waitUntil(timeout: 1) { done in
+                NSThread.sleepForTimeInterval(5.0)
+                done()
+            }
         }
+    }
 
-        failsWithErrorMessage("Stall on main thread - too much enqueued on main run loop before waitUntil executes.") {
-            waitUntil { done in
+    func testCombiningAsyncWaitUntilAndToEventuallyIsNotAllowed() {
+        let referenceLine = __LINE__ + 9
+        var msg = "Unexpected exception raised: Nested async expectations are not allowed "
+        msg += "to avoid creating flaky tests."
+        msg += "\n\n"
+        msg += "The call to\n\t"
+        msg += "expect(...).toEventually(...) at \(__FILE__):\(referenceLine + 7)\n"
+        msg += "triggered this exception because\n\t"
+        msg += "waitUntil(...) at \(__FILE__):\(referenceLine + 1)\n"
+        msg += "is currently managing the main run loop."
+        failsWithErrorMessage(msg) { // reference line
```

Comment:
It's just a line number to make all the other line number indicators derived of. This makes it a bit easier to change the error message without having to update every time the error message changes.

The error message emits line numbers based on where `waitUntil` and `toEventually` is used.